FileShareService.DesktopClient.Core/TokenCacheHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.Identity.Client;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace UKHO.FileShareService.DesktopClient.Core
+{
+    static class TokenCacheHelper
+    {
+        static TokenCacheHelper()
+        {
+            try
+            {
+                // For packaged desktop apps (MSIX packages, also called desktop bridge) the executing assembly folder is read-only. 
+                // In that case we need to use Windows.Storage.ApplicationData.Current.LocalCacheFolder.Path + "\msalcache.bin" 
+                // which is a per-app read/write folder for packaged apps.
+                // See https://docs.microsoft.com/windows/msix/desktop/desktop-to-uwp-behind-the-scenes
+                //  CacheFilePath = Path.Combine(Windows.Storage.ApplicationData.Current.LocalCacheFolder.Path, "msalcache.bin");
+                CacheFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "msalcache.bin");
+                // CacheFilePath = Path.Combine(@"D:\FSSTooling\TestPath", "msalcache.bin");
+            }
+            catch (System.InvalidOperationException)
+            {
+                // Fall back for an unpackaged desktop app
+                CacheFilePath = System.Reflection.Assembly.GetExecutingAssembly().Location + "msalcache.bin";
```

Comment:
The user won't have permissions to write to this directory (it is installed on a shared terminal server). If you can't write to ApplicationData, you could try falling back to MyDocuments
```suggestion
                CacheFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "msalcache.bin");
```

Reply:
Resolved